### PR TITLE
Initialize basic development framework for TrafficManager

### PR DESCRIPTION
**Pull request by ESWZY**
## Description

To initialize basic development framework for TrafficManager.

Ref: #467 

Based on abstractions of Kubernetes Services and Pods, as well as the modification of network request events, this project can achieve the following functionalities through operational logic:

1. Service Resolution: It directs requests straight to backend Pods, bypassing massive iptables lookup and NAT through iptables.

Architecture

2. Non-intrusive Traffic Management: It offers the ability to modify traffic across Pods and Services. Specific callers can invoke particular versions of a service, facilitating traffic migration or version rolling upgrades.
3. Metric-Based Managing: By using metric inputs, it filters and eliminates abnormal Pods, preventing requests from reaching malfunctioning Pods. If all Pods are unable to function correctly, the request is denied outright.

Dynamic Control

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: eBPF_Supermarket/TrafficManager/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.idea
+.vscode
+
+bpf/headers/vmlinux.h
```
Comment (helight) [inline eBPF_Supermarket/TrafficManager/.gitignore:4]: 整个项目要增加自动化流水线，可以编译测试验证程序代码问题，防止把不可用的代码提交上来

Comment (ESWZY) [reply eBPF_Supermarket/TrafficManager/.gitignore:4]: 添加好了

**Comment by helight** [comment]
@LinkinPF 也看一看

File: eBPF_Supermarket/TrafficManager/bpf/connect.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+static __always_inline __be32 ctx_get_dst_ip(const struct bpf_sock_addr *ctx)
+{
+    volatile __u32 dst_ip = ctx->user_ip4;
+    return (__be32)dst_ip;
+}
+
+static __always_inline __be16 ctx_get_dst_port(const struct bpf_sock_addr *ctx)
+{
+	volatile __u32 dport = ctx->user_port;
+	return (__be16)dport;
```
Comment (LinkinPF) [inline eBPF_Supermarket/TrafficManager/bpf/connect.h:26]: 后续可以在IDE装一个自动格式化代码的工具，缩进这些风格就可以统一了

**Review by LinkinPF — APPROVED** [review]

**Pull request merged**